airflow/jobs/local_task_job_runner.py

```diff
@@ -306,3 +306,13 @@ def _log_return_code_metric(self, return_code: int):
             "local_task_job.task_exit."
             f"{self.job.id}.{self.task_instance.dag_id}.{self.task_instance.task_id}.{return_code}"
         )
+        # Same metric with tagging
+        Stats.incr(
```

Comment:
That was one of the ideas I had for the Metrics Registry, actually.   It could be used to record the template used to generate the old names and that template could be used to extract it down into the new names.  Unfortunately there isn't a great deal of consistency on the names at this point so it's not as easy as just doing a split(), sadly.

Reply:
Or maybe it can... if i pass in the new name with the attributes/tags, then in incr() just emit it that way and then emit it concatenated all together... but then I'd have to worry about sorting order of the items in `attributes`which might be a different nightmare...

Maybe adjust incr() so it takes an optional "back-compat-name" parameter?  If it exists, then emit both?

Reply:
> Maybe adjust incr() so it takes an optional "back-compat-name" parameter? If it exists, then emit both?

Then it's up to the caller to know the right name and character limit right?

Reply:
I just saw that last comment.   Yeah, it would be up to whoever is adding a new metric to know about the character limit on the name.  Maybe have a look at the latest implementation of incr() and let me know what you think.  If we go with the current technique, then we can get rid of these double calls all over the place and consolidate them to `incr("new_name, tags=tags, back_compat_name=f'name_{with}_{embedded}_{tags}`

Reply:
If/When we move over to the centralized Registry idea, then a pre-commit could enforce it there

Reply:
Had a chat with Niko and we've decided the redundant calls here are the lesser of the two evils and will make it easier to deprecate the long names at a later date rather than hiding the problem. I'm going to roll back that change in a coming update to this PR and leave all the duplicate calls in place.  Resolving.